app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryTest.kt

```diff
@@ -1,15 +1,84 @@
 package com.lightricks.feedexercise.data
 
+import android.content.Context
+import android.util.Log
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.Observer
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.MockFeedApiService
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
 import org.junit.runner.RunWith
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
 
+
 @RunWith(AndroidJUnit4::class)
 class FeedRepositoryTest {
-   //todo: add the tests here
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var db: FeedDatabase
+    private lateinit var feedApiService: FeedApiService
+
+    @Before
+    fun setup() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        createDb(context)
+        createApiService(context)
+    }
+
+    private fun createDb(context: Context) {
+        db = Room.inMemoryDatabaseBuilder(
+            context, FeedDatabase::class.java
+        ).build()
+    }
+
+    private fun createApiService(context: Context) {
+        feedApiService = MockFeedApiService(context)
+    }
+
+    @Test
```

Comment:
We could also add a test that verifies that `refresh()` passes network errors that `MockFeedApiService` can generate.